Multiplatform/Utility/Intents/SpotlightHelper.swift

```diff
@@ -19,7 +19,10 @@ struct SpotlightHelper {
     static let logger = Logger(subsystem: "io.rfk.ampfin", category: "Spotlight")
     
     static func donate(force: Bool = false) {
-        if Defaults[.lastSpotlightDonation] + waitTime > Date.timeIntervalSinceReferenceDate {
+        let isFirstDonation = Defaults[.lastSpotlightDonation] == 0 && Defaults[.lastSpotlightDonationCompletion] == 0
+        let lastDonationCompleted = isFirstDonation || Defaults[.lastSpotlightDonationCompletion] > Defaults[.lastSpotlightDonation]
```

Comment:
If `isFirstDonation` is true `lastDonationCompleted` will also be true because of the second check, is this desired?

Reply:
Yes. We need `isFirstDonation` mainly because when the user removed all data or just installed this app, both key will contain the default 0, and we need to treat this as last donation is completed to let the following logic to start a new index.